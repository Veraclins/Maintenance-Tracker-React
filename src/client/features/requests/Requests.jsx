import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const Requests = ({ requests, noRequestMessage, isAdmin }) => {
  const pending = requests.filter(request => request.status === 'pending');
  const approved = requests.filter(request => request.status === 'approved');
  const disapproved = requests.filter(request => request.status === 'disapproved');
  const resolved = requests.filter(request => request.status === 'resolved');
  const path = isAdmin ? 'admin' : 'requests';
  return (
    <div className="request_content_layout body_gradient">
      <div className="container">
        <div className="request_header">
          {requests.length !== 0
            ? (
              <p>
                {`These are all ${isAdmin ? 'the' : 'your'}  requests`}
              </p>
            )
            : (
              <div className="no_content">
                <p className="no_requests">{noRequestMessage}</p>
                {!isAdmin
                  ? (
                    <Link
                      to="/requests/create"
                      className="buttons button_default"
                    >
                      MAKE A REQUEST
                    </Link>
                  ) : false}
              </div>
            )}
        </div>
        {pending.length !== 0
          ? (
            <div className="request_category">
              <h3 className="category_name">Pending</h3>
              <div className="span_desc">
                {pending.map(request => (
                  <div className="request_details" key={request.id}>
                    <span className="request_title">{request.title}</span>
                    <span className="request_button">
                      <Link
                        className="buttons button_small"
                        to={`/${path}/${request.id}`}
                      >
                          View
                      </Link>
                    </span>
                  </div>
                ))}
              </div>
              <div className="clear" />
            </div>
          ) : false}
        <div className="clear" />
        {approved.length !== 0
          ? (
            <div className="request_category">
              <h3 className="category_name">Approved</h3>
              <div className="span_desc">
                {approved.map(request => (
                  <div className="request_details" key={request.id}>
                    <span className="request_title">{request.title}</span>
                    <span className="request_button">
                      <Link
                        className="buttons button_small"
                        to={`/${path}/${request.id}`}
                      >
                          View
                      </Link>
                    </span>
                  </div>
                ))}
              </div>
              <div className="clear" />
            </div>
          ) : false}
        <div className="clear" />
        {disapproved.length !== 0
          ? (
            <div className="request_category">
              <h3 className="category_name">Disapproved</h3>
              <div className="span_desc">
                {disapproved.map(request => (
                  <div className="request_details" key={request.id}>
                    <span className="request_title">{request.title}</span>
                    <span className="request_button">
                      <Link
                        className="buttons button_small"
                        to={`/${path}/${request.id}`}
                      >
                          View
                      </Link>
                    </span>
                  </div>
                ))}
              </div>
              <div className="clear" />
            </div>
          ) : false}
        <div className="clear" />
        {resolved.length !== 0
          ? (
            <div className="request_category">
              <h3 className="category_name">Resolved</h3>
              <div className="span_desc">
                {resolved.map(request => (
                  <div className="request_details" key={request.id}>
                    <span className="request_title">{request.title}</span>
                    <span className="request_button">
                      <Link
                        className="buttons button_small"
                        to={`/${path}/${request.id}`}
                      >
                          View
                      </Link>
                    </span>
                  </div>
                ))}
              </div>
              <div className="clear" />
            </div>
          ) : false}
        <div className="clear" />
      </div>
    </div>
  );
};

Requests.propTypes = {
  requests: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  isAdmin: PropTypes.bool.isRequired,
  noRequestMessage: PropTypes.string.isRequired,
};

export default Requests;
