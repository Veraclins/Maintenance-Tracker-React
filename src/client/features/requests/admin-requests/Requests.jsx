import React from 'react';

const Requests = () => (
  <section>
    <div>
      <div className="team_banner_layout">
        <div className="image_banner fade-in">
          <div className="container">
            <div className="banner_inner wow fadeInUp" style={{ visibility: 'visible', animationName: 'fadeInUp' }}>
              <h1>Join as a Senior Developer</h1>
              <h5>Named the #1 best place to work in Africa, Andela employs leading technologists who are passionate about using code to drive business results and catalyze change around the world.</h5>
            </div>
          </div>
          <div className="blue_banner_bg overlay_gradient" />
          <div className="blue_dot_bg" />
        </div>
      </div>
      <div className="team_content_layout">
        <div className="container">
          <div className="content_inner">
            <div className="content_block con1 wow fadeInDown" data-wow-delay="0.2s" style={{ visibility: 'visible', animationDelay: '0.2s', animationName: 'fadeInDown' }}>
              <h3>Work with global teams</h3>
              <p>Andela is founded on a simple truth; that brilliance is evenly distributed. We unlock that brilliance by giving global companies access to untapped talent across Africa</p>
              <div className="teamlink"><a href="https://andela.com/partner-with-andela/our-featured-partners/" className="link_blue">FEATURED PARTNERS</a>
              </div>
            </div>
            <div className="content_block con1 wow fadeInDown" data-wow-delay="0.2s" style={{ visibility: 'visible', animationDelay: '0.2s', animationName: 'fadeInDown' }}>
              <h3>Be a catalyst for change</h3>
              <p>We believe in developing our developers, and are looking for leaders who can model, capture, and transfer the mindset and best practices of their craft to any technology professional.</p>
              <div className="teamlink"><a href="https://andela.com/community/" className="link_blue">COMMUNITY/ANDELA OPEN</a>
              </div>
            </div>
            <div className="content_block con1 wow fadeInDown" data-wow-delay="0.2s" style={{ visibility: 'visible', animationDelay: '0.2s', animationName: 'fadeInDown' }}>
              <h3>Advance your career</h3>
              <p>Continually level up on your skills and grow as you help others to grow.</p>
              <div className="teamlink"><a href="https://andela.com/our-developer-framework/" className="link_blue">DEVELOPING DEVELOPERS</a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div className="name_layout body_gradient">
        <div className="container">
          <div className="name_inner">
            <div className="slick-prev slick-arrow" />
            <div className="slick-next slick-arrow" />
            <div className="name_slick_slider slick-initialized slick-slider">
              <button className="slick-prev slick-arrow slick-disabled" aria-label="Previous" type="button" aria-disabled="true" style={{ display: 'block' }}>Previous</button>
              <div className="slick-list draggable">
                <div className="slick-track" style={{ opacity: 1, width: '2130px', transform: 'translate3d(0px, 0px, 0px)' }}>
                  <div className="slick-slide slick-current slick-active" data-slick-index="0" aria-hidden="false" style={{ width: '355px' }}>
                    <div>
                      <div className="name_block" style={{ width: '100%', display: 'inline-block' }}>
                        <div className="nameimg">
                          <img src="https://andela.com/wp-content/uploads/2018/08/Aretha-Kebirungi.png" alt="name1" />
                        </div>
                        <div className="desc_part">
                          <h3>Aretha Kebirungi</h3>
                          <p className="innername">Senior Developer - Uganda</p>
                          <p>“I saw an opportunity to pursue my dream of mentoring upcoming developers to be better and reach places they never imagined they could”</p>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div className="slick-slide slick-active" data-slick-index="1" aria-hidden="false" style={{ width: '355px' }}>
                    <div>
                      <div className="name_block" style={{ width: '100%', display: 'inline-block' }}>
                        <div className="nameimg">
                          <img src="https://andela.com/wp-content/uploads/2018/08/Ebot-Tabi.png" alt="name1" />
                        </div>
                        <div className="desc_part">
                          <h3>Ebot Tabi</h3>
                          <p className="innername">Senior Developer - Cameroon</p>
                          <p>“Working at Andela provides a platform to contribute to the world in a way that positively impacts society with the application &amp; technology I am working on.”</p>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div className="slick-slide" data-slick-index="2" aria-hidden="true" style={{ width: '355px' }} tabIndex="-1">
                    <div>
                      <div className="name_block" style={{ width: '100%', display: 'inline-block' }}>
                        <div className="nameimg">
                          <img src="https://andela.com/wp-content/uploads/2018/08/James-Maina.png" alt="name1" />
                        </div>
                        <div className="desc_part">
                          <h3>James Maina</h3>
                          <p className="innername">Senior Developer - Kenya</p>
                          <p>“The developers at Andela are working on cool projects for big companies. I think that’s how we’ll get Africa on the world tech map.”</p>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div className="slick-slide" data-slick-index="3" aria-hidden="true" style={{ width: '355px' }} tabIndex="-1">
                    <div>
                      <div className="name_block" style={{ width: '100%', display: 'inline-block' }}>
                        <div className="nameimg">
                          <img src="https://andela.com/wp-content/uploads/2018/08/omotayo-madein.png" alt="name1" />
                        </div>
                        <div className="desc_part">
                          <h3>Omotayo Madein</h3>
                          <p className="innername">Senior Developer - Nigeria</p>
                          <p>“I plan to serve in an advisory role for innovative startups that improve human experiences. Working at Andela re-fuels my motivation towards this goal”</p>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div className="slick-slide" data-slick-index="4" aria-hidden="true" style={{ width: '355px' }} tabIndex="-1">
                    <div>
                      <div className="name_block" style={{ width: '100%', display: 'inline-block' }}>
                        <div className="nameimg">
                          <img src="https://andela.com/wp-content/uploads/2018/08/Mohamed-Omar.png" alt="name1" />
                        </div>
                        <div className="desc_part">
                          <h3>Mohamed Omar</h3>
                          <p className="innername">Technical Success Manager - Ke...</p>
                          <p>“Within one or two generations, we can turn Africa into the top of the tech world, and I believe this is where it will start.”</p>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div className="slick-slide" data-slick-index="5" aria-hidden="true" style={{ width: '355px' }} tabIndex="-1">
                    <div>
                      <div className="name_block" style={{ width: '100%', display: 'inline-block' }}>
                        <div className="nameimg">
                          <img src="https://andela.com/wp-content/uploads/2018/08/Adegbenga-Adeye.png" alt="name1" />
                        </div>
                        <div className="desc_part">
                          <h3>Adegbenga Adeye</h3>
                          <p className="innername">Technical Success Manager - Ni...</p>
                          <p>“Andela encourages me to push developers to push themselves. It’s the best way they can learn, and Andelans understand that.”</p>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <button className="slick-next slick-arrow" aria-label="Next" type="button" style={{ display: 'block' }} aria-disabled="false">Next</button>
            </div>
          </div>
          <div className="name_background" />
        </div>
      </div>
      <div className="button_career_bottom_layout">
        <div className="container">
          <p>Not sure where to start?</p>
          <a href="#form-modal-stayupdated" className="open-popup-link buttons button_default wow fadeInLeft bottom_button" data-wow-delay="0.5s" style={{ visibility: 'visible', animationDelay: '0.5s', animationName: 'fadeInLeft' }}>GET STARTED HERE</a>
        </div>
      </div>
    </div>
  </section>
);

export default Requests;
