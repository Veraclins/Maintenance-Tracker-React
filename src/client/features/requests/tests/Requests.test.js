import React from 'react';
import { shallow } from 'enzyme';

import { Requests } from '../Requests';

const user = {
  firstName: 'Clinton',
  lastName: 'Agada',
};
const location = {
  pathname: '',
};
const userReq = [
  {
    id: 1,
    user_id: 2,
    title: 'Excellent Work',
    device: 'Desktop',
    description: 'Although a downpour briefly interrupted play in the early stages of the second half, Nigeria played with more intent after half-time and went close in the 63rd minute, as John Obi Mikel saw a header saved by Vaclik.',
    status: 'disapproved',
    created_at: '2018-10-06T23:14:09.748Z',
    updated_at: '2018-10-06T23:14:25.357Z',
  },
  {
    id: 13,
    user_id: 2,
    title: 'General repainting',
    device: 'Smartphone',
    description: 'Although a downpour briefly interrupted play in the early stages of the second half, Nigeria played with more intent after half-time and went close in the 63rd minute, as John Obi Mikel saw a header saved by Vaclik.',
    status: 'pending',
    created_at: '2018-10-06T23:14:25.294Z',
    updated_at: '2018-10-06T23:14:25.294Z',
  },
  {
    id: 12,
    user_id: 2,
    title: 'General repainting',
    device: 'Smartphone',
    description: 'Although a downpour briefly interrupted play in the early stages of the second half, Nigeria played with more intent after half-time and went close in the 63rd minute, as John Obi Mikel saw a header saved by Vaclik.',
    status: 'pending',
    created_at: '2018-10-06T23:14:10.132Z',
    updated_at: '2018-10-06T23:14:10.132Z',
  },
  {
    id: 4,
    user_id: 2,
    title: 'Routine Maintenance',
    device: 'Smartphone',
    description: 'If you are passing parameters to your queries you will want to avoid string concatenating parameters into the query text directly. This can lead to sql injection vulnerabilities. node-postgres supports paramterized queries, passing your query text unaltered as well as your parameters to the PostgreSQL server where the parameters are safely substituted into the query with battle-tested parameter substitution code within the server itself.',
    status: 'resolved',
    created_at: '2018-10-06T23:14:09.748Z',
    updated_at: '2018-10-06T23:14:09.985Z',
  },
  {
    id: 7,
    user_id: 4,
    title: 'Routine Maintenance',
    device: 'Desktop',
    description: 'If you are passing parameters to your queries you will want to avoid string concatenating parameters into the query text directly. This can lead to sql injection vulnerabilities. node-postgres supports paramterized queries, passing your query text unaltered as well as your parameters to the PostgreSQL server where the parameters are safely substituted into the query with battle-tested parameter substitution code within the server itself.',
    status: 'pending',
    created_at: '2018-10-06T23:14:09.748Z',
    updated_at: '2018-10-06T23:14:09.748Z',
  },
  {
    id: 8,
    user_id: 3,
    title: 'Routine Maintenance',
    device: 'Tablet',
    description: 'If you are passing parameters to your queries you will want to avoid string concatenating parameters into the query text directly. This can lead to sql injection vulnerabilities. node-postgres supports paramterized queries, passing your query text unaltered as well as your parameters to the PostgreSQL server where the parameters are safely substituted into the query with battle-tested parameter substitution code within the server itself.',
    status: 'pending',
    created_at: '2018-10-06T23:14:09.748Z',
    updated_at: '2018-10-06T23:14:09.748Z',
  },
];
const adminReq = [
  {
    id: 1,
    user_id: 2,
    title: 'Excellent Work',
    device: 'Desktop',
    description: 'Although a downpour briefly interrupted play in the early stages of the second half, Nigeria played with more intent after half-time and went close in the 63rd minute, as John Obi Mikel saw a header saved by Vaclik.',
    status: 'disapproved',
    created_at: '2018-10-06T23:14:09.748Z',
    updated_at: '2018-10-06T23:14:25.357Z',
  },
  {
    id: 13,
    user_id: 2,
    title: 'General repainting',
    device: 'Smartphone',
    description: 'Although a downpour briefly interrupted play in the early stages of the second half, Nigeria played with more intent after half-time and went close in the 63rd minute, as John Obi Mikel saw a header saved by Vaclik.',
    status: 'pending',
    created_at: '2018-10-06T23:14:25.294Z',
    updated_at: '2018-10-06T23:14:25.294Z',
  },
  {
    id: 12,
    user_id: 2,
    title: 'General repainting',
    device: 'Smartphone',
    description: 'Although a downpour briefly interrupted play in the early stages of the second half, Nigeria played with more intent after half-time and went close in the 63rd minute, as John Obi Mikel saw a header saved by Vaclik.',
    status: 'pending',
    created_at: '2018-10-06T23:14:10.132Z',
    updated_at: '2018-10-06T23:14:10.132Z',
  },
  {
    id: 4,
    user_id: 2,
    title: 'Routine Maintenance',
    device: 'Smartphone',
    description: 'If you are passing parameters to your queries you will want to avoid string concatenating parameters into the query text directly. This can lead to sql injection vulnerabilities. node-postgres supports paramterized queries, passing your query text unaltered as well as your parameters to the PostgreSQL server where the parameters are safely substituted into the query with battle-tested parameter substitution code within the server itself.',
    status: 'resolved',
    created_at: '2018-10-06T23:14:09.748Z',
    updated_at: '2018-10-06T23:14:09.985Z',
  },
  {
    id: 7,
    user_id: 4,
    title: 'Routine Maintenance',
    device: 'Desktop',
    description: 'If you are passing parameters to your queries you will want to avoid string concatenating parameters into the query text directly. This can lead to sql injection vulnerabilities. node-postgres supports paramterized queries, passing your query text unaltered as well as your parameters to the PostgreSQL server where the parameters are safely substituted into the query with battle-tested parameter substitution code within the server itself.',
    status: 'pending',
    created_at: '2018-10-06T23:14:09.748Z',
    updated_at: '2018-10-06T23:14:09.748Z',
  },
  {
    id: 8,
    user_id: 3,
    title: 'Routine Maintenance',
    device: 'Tablet',
    description: 'If you are passing parameters to your queries you will want to avoid string concatenating parameters into the query text directly. This can lead to sql injection vulnerabilities. node-postgres supports paramterized queries, passing your query text unaltered as well as your parameters to the PostgreSQL server where the parameters are safely substituted into the query with battle-tested parameter substitution code within the server itself.',
    status: 'pending',
    created_at: '2018-10-06T23:14:09.748Z',
    updated_at: '2018-10-06T23:14:09.748Z',
  },
  {
    id: 2,
    user_id: 2,
    title: 'Browsers Crash Always',
    device: 'Laptop',
    description: 'The quarterly routine maintenance service for the Elepaq 3.5KVA Generator is long overdue and necessary in order to forestall total breakdown. The love of the lord is the beginning of wisdom.',
    status: 'pending',
    created_at: '2018-10-06T23:14:09.748Z',
    updated_at: '2018-10-06T23:14:09.748Z',
  },
  {
    id: 10,
    user_id: 3,
    title: 'Routine Maintenance',
    device: 'Smartphone',
    description: 'If you are passing parameters to your queries you will want to avoid string concatenating parameters into the query text directly. This can lead to sql injection vulnerabilities. node-postgres supports paramterized queries, passing your query text unaltered as well as your parameters to the PostgreSQL server where the parameters are safely substituted into the query with battle-tested parameter substitution code within the server itself.',
    status: 'pending',
    created_at: '2018-10-06T23:14:09.748Z',
    updated_at: '2018-10-06T23:14:09.748Z',
  },
  {
    id: 11,
    user_id: 2,
    title: 'Battery Problem',
    device: 'Laptop',
    description: 'If you are passing parameters to your queries you will want to avoid string concatenating parameters into the query text directly. This can lead to sql injection vulnerabilities. node-postgres supports paramterized queries, passing your query text unaltered as well as your parameters to the PostgreSQL server where the parameters are safely substituted into the query with battle-tested parameter substitution code within the server itself.',
    status: 'pending',
    created_at: '2018-10-06T23:14:09.748Z',
    updated_at: '2018-10-06T23:14:09.748Z',
  },
  {
    id: 9,
    user_id: 3,
    title: 'Faulty Touch Pad',
    device: 'Smartphone',
    description: 'If you are passing parameters to your queries you will want to avoid string concatenating parameters into the query text directly. This can lead to sql injection vulnerabilities. node-postgres supports paramterized queries, passing your query text unaltered as well as your parameters to the PostgreSQL server where the parameters are safely substituted into the query with battle-tested parameter substitution code within the server itself.',
    status: 'pending',
    created_at: '2018-10-06T23:14:09.748Z',
    updated_at: '2018-10-06T23:14:09.748Z',
  },
  {
    id: 3,
    user_id: 2,
    title: 'Screen Problems',
    device: 'Tablet',
    description: 'If you are passing parameters to your queries you will want to avoid string concatenating parameters into the query text directly. This can lead to sql injection vulnerabilities. node-postgres supports paramterized queries, passing your query text unaltered as well as your parameters to the PostgreSQL server where the parameters are safely substituted into the query with battle-tested parameter substitution code within the server itself.',
    status: 'pending',
    created_at: '2018-10-06T23:14:09.748Z',
    updated_at: '2018-10-06T23:14:09.748Z',
  },
  {
    id: 5,
    user_id: 2,
    title: 'Routine Maintenance',
    device: 'Laptop',
    description: 'If you are passing parameters to your queries you will want to avoid string concatenating parameters into the query text directly. This can lead to sql injection vulnerabilities. node-postgres supports paramterized queries, passing your query text unaltered as well as your parameters to the PostgreSQL server where the parameters are safely substituted into the query with battle-tested parameter substitution code within the server itself.',
    status: 'pending',
    created_at: '2018-10-06T23:14:09.748Z',
    updated_at: '2018-10-06T23:14:09.748Z',
  },
  {
    id: 6,
    user_id: 3,
    title: 'Intermitent freezing of apps',
    device: 'Smartphone',
    description: 'If you are passing parameters to your queries you will want to avoid string concatenating parameters into the query text directly. This can lead to sql injection vulnerabilities. node-postgres supports paramterized queries, passing your query text unaltered as well as your parameters to the PostgreSQL server where the parameters are safely substituted into the query with battle-tested parameter substitution code within the server itself.',
    status: 'pending',
    created_at: '2018-10-06T23:14:09.748Z',
    updated_at: '2018-10-06T23:14:09.748Z',
  },
];
const mockFunction = jest.fn();
const noRequestMessage = "You don't have any request yet";

describe('Test the Requests component', () => {
  it('renders with the right view for a logged in user with no request', () => {
    const isLoggedIn = true;
    const isAdmin = false;
    const requests = [];
    const wrapper = shallow(<Requests
      requests={requests}
      isAdmin={isAdmin}
      getRequests={mockFunction}
      user={user}
      location={location}
      isLoggedIn={isLoggedIn}
      noRequestMessage={noRequestMessage}
    />);
    expect(wrapper.find('div')).toHaveLength(8);
    expect(wrapper).toMatchSnapshot();
  });
  it('renders with the right view for a logged in user with some requests', () => {
    const isLoggedIn = true;
    const isAdmin = false;
    const requests = userReq;
    const wrapper = shallow(<Requests
      requests={requests}
      isAdmin={isAdmin}
      getRequests={mockFunction}
      user={user}
      location={location}
      isLoggedIn={isLoggedIn}
      noRequestMessage={noRequestMessage}
    />);
    expect(wrapper.find('div')).toHaveLength(22);
    expect(wrapper).toMatchSnapshot();
  });
  it('renders with the right view for a logged in user', () => {
    const isLoggedIn = true;
    const isAdmin = true;
    const requests = adminReq;
    const wrapper = shallow(<Requests
      requests={requests}
      isAdmin={isAdmin}
      getRequests={mockFunction}
      user={user}
      location={location}
      isLoggedIn={isLoggedIn}
      noRequestMessage={noRequestMessage}
    />);
    expect(wrapper.find('div')).toHaveLength(29);
    expect(wrapper).toMatchSnapshot();
  });
});
