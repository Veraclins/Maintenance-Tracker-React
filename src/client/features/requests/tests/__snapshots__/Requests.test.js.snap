// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Requests component renders with the right view for a logged in user 1`] = `
<div
  className="request_content_layout body_gradient"
>
  <div
    className="container"
  >
    <div
      className="request_header"
    >
      <p>
        These are all the  requests
      </p>
    </div>
    <div
      className="request_category"
    >
      <h3
        className="category_name"
      >
        Pending
      </h3>
      <div
        className="span_desc"
      >
        <div
          className="request_details"
          key="13"
        >
          <span
            className="request_title"
          >
            General repainting
          </span>
          <span
            className="request_button"
          >
            <Link
              className="buttons button_small"
              replace={false}
              to="/admin/13"
            >
              View
            </Link>
          </span>
        </div>
        <div
          className="request_details"
          key="12"
        >
          <span
            className="request_title"
          >
            General repainting
          </span>
          <span
            className="request_button"
          >
            <Link
              className="buttons button_small"
              replace={false}
              to="/admin/12"
            >
              View
            </Link>
          </span>
        </div>
        <div
          className="request_details"
          key="7"
        >
          <span
            className="request_title"
          >
            Routine Maintenance
          </span>
          <span
            className="request_button"
          >
            <Link
              className="buttons button_small"
              replace={false}
              to="/admin/7"
            >
              View
            </Link>
          </span>
        </div>
        <div
          className="request_details"
          key="8"
        >
          <span
            className="request_title"
          >
            Routine Maintenance
          </span>
          <span
            className="request_button"
          >
            <Link
              className="buttons button_small"
              replace={false}
              to="/admin/8"
            >
              View
            </Link>
          </span>
        </div>
        <div
          className="request_details"
          key="2"
        >
          <span
            className="request_title"
          >
            Browsers Crash Always
          </span>
          <span
            className="request_button"
          >
            <Link
              className="buttons button_small"
              replace={false}
              to="/admin/2"
            >
              View
            </Link>
          </span>
        </div>
        <div
          className="request_details"
          key="10"
        >
          <span
            className="request_title"
          >
            Routine Maintenance
          </span>
          <span
            className="request_button"
          >
            <Link
              className="buttons button_small"
              replace={false}
              to="/admin/10"
            >
              View
            </Link>
          </span>
        </div>
        <div
          className="request_details"
          key="11"
        >
          <span
            className="request_title"
          >
            Battery Problem
          </span>
          <span
            className="request_button"
          >
            <Link
              className="buttons button_small"
              replace={false}
              to="/admin/11"
            >
              View
            </Link>
          </span>
        </div>
        <div
          className="request_details"
          key="9"
        >
          <span
            className="request_title"
          >
            Faulty Touch Pad
          </span>
          <span
            className="request_button"
          >
            <Link
              className="buttons button_small"
              replace={false}
              to="/admin/9"
            >
              View
            </Link>
          </span>
        </div>
        <div
          className="request_details"
          key="3"
        >
          <span
            className="request_title"
          >
            Screen Problems
          </span>
          <span
            className="request_button"
          >
            <Link
              className="buttons button_small"
              replace={false}
              to="/admin/3"
            >
              View
            </Link>
          </span>
        </div>
        <div
          className="request_details"
          key="5"
        >
          <span
            className="request_title"
          >
            Routine Maintenance
          </span>
          <span
            className="request_button"
          >
            <Link
              className="buttons button_small"
              replace={false}
              to="/admin/5"
            >
              View
            </Link>
          </span>
        </div>
        <div
          className="request_details"
          key="6"
        >
          <span
            className="request_title"
          >
            Intermitent freezing of apps
          </span>
          <span
            className="request_button"
          >
            <Link
              className="buttons button_small"
              replace={false}
              to="/admin/6"
            >
              View
            </Link>
          </span>
        </div>
      </div>
      <div
        className="clear"
      />
    </div>
    <div
      className="clear"
    />
    <div
      className="clear"
    />
    <div
      className="request_category"
    >
      <h3
        className="category_name"
      >
        Disapproved
      </h3>
      <div
        className="span_desc"
      >
        <div
          className="request_details"
          key="1"
        >
          <span
            className="request_title"
          >
            Excellent Work
          </span>
          <span
            className="request_button"
          >
            <Link
              className="buttons button_small"
              replace={false}
              to="/admin/1"
            >
              View
            </Link>
          </span>
        </div>
      </div>
      <div
        className="clear"
      />
    </div>
    <div
      className="clear"
    />
    <div
      className="request_category"
    >
      <h3
        className="category_name"
      >
        Resolved
      </h3>
      <div
        className="span_desc"
      >
        <div
          className="request_details"
          key="4"
        >
          <span
            className="request_title"
          >
            Routine Maintenance
          </span>
          <span
            className="request_button"
          >
            <Link
              className="buttons button_small"
              replace={false}
              to="/admin/4"
            >
              View
            </Link>
          </span>
        </div>
      </div>
      <div
        className="clear"
      />
    </div>
    <div
      className="clear"
    />
  </div>
</div>
`;

exports[`Test the Requests component renders with the right view for a logged in user with no request 1`] = `
<div
  className="request_content_layout body_gradient"
>
  <div
    className="container"
  >
    <div
      className="request_header"
    >
      <div
        className="no_content"
      >
        <p
          className="no_requests"
        >
          You don't have any request yet
        </p>
        <Link
          className="buttons button_default"
          replace={false}
          to="/requests/create"
        >
          MAKE A REQUEST
        </Link>
      </div>
    </div>
    <div
      className="clear"
    />
    <div
      className="clear"
    />
    <div
      className="clear"
    />
    <div
      className="clear"
    />
  </div>
</div>
`;

exports[`Test the Requests component renders with the right view for a logged in user with some requests 1`] = `
<div
  className="request_content_layout body_gradient"
>
  <div
    className="container"
  >
    <div
      className="request_header"
    >
      <p>
        These are all your  requests
      </p>
    </div>
    <div
      className="request_category"
    >
      <h3
        className="category_name"
      >
        Pending
      </h3>
      <div
        className="span_desc"
      >
        <div
          className="request_details"
          key="13"
        >
          <span
            className="request_title"
          >
            General repainting
          </span>
          <span
            className="request_button"
          >
            <Link
              className="buttons button_small"
              replace={false}
              to="/requests/13"
            >
              View
            </Link>
          </span>
        </div>
        <div
          className="request_details"
          key="12"
        >
          <span
            className="request_title"
          >
            General repainting
          </span>
          <span
            className="request_button"
          >
            <Link
              className="buttons button_small"
              replace={false}
              to="/requests/12"
            >
              View
            </Link>
          </span>
        </div>
        <div
          className="request_details"
          key="7"
        >
          <span
            className="request_title"
          >
            Routine Maintenance
          </span>
          <span
            className="request_button"
          >
            <Link
              className="buttons button_small"
              replace={false}
              to="/requests/7"
            >
              View
            </Link>
          </span>
        </div>
        <div
          className="request_details"
          key="8"
        >
          <span
            className="request_title"
          >
            Routine Maintenance
          </span>
          <span
            className="request_button"
          >
            <Link
              className="buttons button_small"
              replace={false}
              to="/requests/8"
            >
              View
            </Link>
          </span>
        </div>
      </div>
      <div
        className="clear"
      />
    </div>
    <div
      className="clear"
    />
    <div
      className="clear"
    />
    <div
      className="request_category"
    >
      <h3
        className="category_name"
      >
        Disapproved
      </h3>
      <div
        className="span_desc"
      >
        <div
          className="request_details"
          key="1"
        >
          <span
            className="request_title"
          >
            Excellent Work
          </span>
          <span
            className="request_button"
          >
            <Link
              className="buttons button_small"
              replace={false}
              to="/requests/1"
            >
              View
            </Link>
          </span>
        </div>
      </div>
      <div
        className="clear"
      />
    </div>
    <div
      className="clear"
    />
    <div
      className="request_category"
    >
      <h3
        className="category_name"
      >
        Resolved
      </h3>
      <div
        className="span_desc"
      >
        <div
          className="request_details"
          key="4"
        >
          <span
            className="request_title"
          >
            Routine Maintenance
          </span>
          <span
            className="request_button"
          >
            <Link
              className="buttons button_small"
              replace={false}
              to="/requests/4"
            >
              View
            </Link>
          </span>
        </div>
      </div>
      <div
        className="clear"
      />
    </div>
    <div
      className="clear"
    />
  </div>
</div>
`;
